src/launcher/java/org/truffleruby/launcher/RubyLauncher.java

```diff
@@ -173,6 +176,47 @@ protected AbortException abortUnrecognizedArgument(String argument) {
                 "truffleruby: invalid option " + argument + "  (Use --help for usage instructions.)");
     }
 
+    @Override
+    protected boolean parseCommonOption(String defaultOptionPrefix, Map<String, String> polyglotOptions,
+            boolean experimentalOptions, String arg) {
+        if (arg.startsWith("--help")) {
+            helpOptionUsed = true;
+        }
+
+        return super.parseCommonOption(defaultOptionPrefix, polyglotOptions, experimentalOptions, arg);
+    }
```

Comment:
@eregon Since I can't use `config` in the `runLauncherAction` I decide to take the same approach as in `graal` and set the flag

Reply:
Why not? `config` is a field of `RubyLauncher` as well.

Reply:
I see, it's a convenient way to detect any `--help*` option, and keeps the `--help` handling like it is.